entity_collection/munge/mongo_import/entities/ranking_metrics/RelevanceCounter.py

```diff
@@ -88,33 +60,35 @@ def db_connect(self):
             self.dbpath = os.path.join(os.path.dirname(__file__), 'db', self.name + ".db")
             self.db = slt.connect(self.dbpath)
         
-    def get_raw_relevance_metrics(self, uri, entity):
-        self.db_connect()
-        csr = self.db.cursor()
-        csr.execute("SELECT id, wikidata_id, wikipedia_hits, europeana_enrichment_hits, europeana_string_hits, pagerank FROM hits WHERE id='"+ uri + "'")
-        first_row = csr.fetchone()
+    def get_raw_relevance_metrics(self, uri, entity, importer):
+        first_row = None
+        try:
+            self.db_connect()
+            csr = self.db.cursor()
+            csr.execute("SELECT id, wikidata_id, wikipedia_hits, europeana_enrichment_hits, europeana_string_hits, pagerank FROM hits WHERE id='"+ uri + "'")
+            first_row = csr.fetchone()
+        except:
+            # database is empty - calculate metrics
+            wikidata_id = self.extract_wikidata_uri(entity)
+            metrics = importer.import_metrics_for_entity(uri, wikidata_id)
+            return metrics
+        
         if(first_row is not None):
             (_, wikidata_id, wikipedia_hits, europeana_enrichment_hits, europeana_string_hits, pagerank) = first_row
             if(pagerank is None):
                 pagerank = 0
-        else:
+        else:            
             # wikipedia_hits is not used anymore
             wikipedia_hits = -1 
             europeana_enrichment_hits = self.get_enrichment_count(uri)
             europeana_string_hits = self.get_label_count(entity['representation'])
             wikidata_id = self.extract_wikidata_uri(entity)
             #TODO import page rank to DB file
-            #TODO use MetricRecord object
+            #TODO use MetricsRecord object
             pagerank = 0.0
             csr.execute("INSERT INTO hits(id, wikidata_id, wikipedia_hits, europeana_enrichment_hits, europeana_string_hits, pagerank) VALUES (?, ?, ?, ?, ?, ?)", (uri, wikidata_id, wikipedia_hits, europeana_enrichment_hits, europeana_string_hits, pagerank))
             self.db.commit()
-        metrics = {
-            self.METRIC_WIKI_HITS : wikipedia_hits,
-            self.METRIC_ENRICHMENT_HITS : europeana_enrichment_hits,
-            self.METRIC_TERM_HITS : europeana_string_hits,
-            self.METRIC_PAGERANK : pagerank
-        }
-        return metrics
+        return MetricsRecord(uri, 'fake label', wikidata_id, wikipedia_hits, europeana_string_hits, europeana_enrichment_hits, pagerank)
```

Comment:
please use the extract_def_label method from MetricsImporter to extract the default label. The MetricsImporter.extract_def_label can be updated to take entity as param and to get read of the edm_country

Reply:
refactored as required